Frontend/Pages/UserManagement/Register.razor

```diff
@@ -83,14 +87,28 @@
     </div>
 </div>
 
-@* TODO implement registration logic *@
 
 @code {
     readonly Registration _registration = new();
 
-    private void HandleValidSubmit() {
-        Console.WriteLine("Valid");
-        Console.WriteLine(_registration.Email);
+    private async void HandleValidSubmit() {
+        var request = new HttpRequestMessage(HttpMethod.Post,
+            "https://localhost:7038/api/register");
+
+        request.Content = new StringContent(JsonConvert.SerializeObject(_registration), Encoding.UTF8, "application/json");
+
+        var client = ClientFactory.CreateClient();
+
+        var response = await client.SendAsync(request);
+
+        if (response.IsSuccessStatusCode)
+        {
+            NavManager.NavigateTo("/");
+        }
+        else
+        {
+            NavManager.NavigateTo("/500");
```

Comment:
This should probably cover more cases than Success and everything else. I don't think the backend is a whole lot more specific, but e.g. when entering a mail that is alredy mapped to an user, the user should probably be informed of that.

Reply:
Pretty sure you can return an error body, might wanna look into returning an error code in JSON or directly.